### PR TITLE
[back/front] Reorganize settings for entity types (#2683)

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Proposed changes

* Move Status Templates to Labels & Attributes
* Rename Workflow tab with entity type
* Re-work entity type menu
* Re-align setting enforce reference & hidden entity in all the platform
* * Use hook to check the setting enforce reference -> migrate all the concerned component
* * Use hook to check the setting hidden entity -> migrate all the concerned component
* Fix a major bug on enforce references with backward compatibility and some minor bugs

 

### Related issues

* [Issue 2683](https://github.com/OpenCTI-Platform/opencti/issues/2683)

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/domain/subType.js [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+import * as R from 'ramda';
+import { ABSTRACT_STIX_CORE_RELATIONSHIP, ABSTRACT_STIX_DOMAIN_OBJECT, schemaTypes } from '../schema/general';
+import { STIX_SIGHTING_RELATIONSHIP } from '../schema/stixSightingRelationship';
+import { buildPagination } from '../database/utils';
 
 export const findById = (subTypeId) => querySubType(subTypeId);
 
 export const findAll = (args) => querySubTypes(args);
+
+export const querySubType = async (subTypeId) => {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/domain/subType.js:10]: Not sure to understand this part ?

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting-resolvers.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  Query: {
+    entitySetting: (_, { id }, context) => findById(context, context.user, id),
+    entitySettings: (_, args, context) => findAll(context, context.user, args),
+    entitySettingByType: (_, { targetType }, context) => findByType(context, context.user, targetType),
+  },
+  Mutation: {
+    entitySettingFieldPatch: (_, { id, input }, context) => {
+      return entitySettingEditField(context, context.user, id, input);
+    },
+    entitySettingsFieldPatch: (_, { ids, input }, context) => {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting-resolvers.ts:11]: Not sure why we need two mutation to do quite the same thing ? Single update = Array update with only one id ?

File: opencti-platform/opencti-graphql/tests/utils/testQuery.ts [hunk 3]
```diff
@@ -80,3 +80,12 @@ export const serverFromUser = (user = ADMIN_USER) => {
 
 const adminApolloServer = serverFromUser();
 export const queryAsAdmin = (request: any) => adminApolloServer.executeOperation(request);
+
+export const isSorted = (arr: []) => {
```
Comment (Kedae) [inline opencti-platform/opencti-graphql/tests/utils/testQuery.ts:84]: ⁉️

File: opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  },
+}));
+
+interface Status {
+  id: string
+  order: number
+  template: {
+    color: string
+    name: string
+  } | null
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx:36]: Why can you have a null item in the list ?

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionContainer.js [hunk 5]
```diff
@@ -65,7 +66,7 @@ const IndividualEditionContainer = (props) => {
       <div className={classes.container}>
         <IndividualEditionOverview
           individual={individual}
-          enableReferences={enableReferences}
+          enableReferences={useIsEnforceReference('Individual')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionContainer.js:69]: This is working ?

File: opencti-platform/opencti-graphql/src/domain/subType.js (hunk 1 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/src/domain/subType.js:10]: I just moved the querySubTypes method from middleware.js to this file.

File: opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting-resolvers.ts (hunk 2 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting-resolvers.ts:11]: You are perfectly right !
I will changed it

File: opencti-platform/opencti-graphql/tests/utils/testQuery.ts (hunk 3 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/tests/utils/testQuery.ts:84]: Just an helper to check if an array is sorted.

File: opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx (hunk 4 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx:36]: I don't have the full knowledge about it. Maybe @Archidoit could help us.

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionContainer.js (hunk 5 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionContainer.js:69]: Yes, why ?

File: opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx (hunk 4 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx:36]: Indeed a template can be null everywhere... maybe that's something we should modify

Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/components/ItemStatusTemplate.tsx:36]: A status is not valid without a template If i'm understand the functionnality properly. Right ?

**Comment by RomuDeuxfois** [comment]
It could be great to add the incident case entity to this new system (mechanism to add and to enforce reference) @SamuelHassine .

Let me know if you want me to do it.

**Pull request merged**